add-ons/pmpro-subscription-delays/subscription-delay-one-time-credit-card-required-trial.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * This code stores data when a user checks out for a level.
+ * If that user tries to checkout for the same level, the Subscription Delay is removed.
+ * The user is instead charged for their first subscription payment at checkout.
+ *
+ * title: One time credit card required trial
+ * layout: snippet
+ * collection: add-ons, pmpro-subscription-delays
+ * category: subscriptions, trials
+ * link: https://www.paidmembershipspro.com/subscription-delay-one-time-credit-card-required-trial/
+ *
+ * You can add this recipe to your site by creating a custom plugin
+ * or using the Code Snippets plugin available for free in the WordPress repository.
+ * Read this companion article for step-by-step directions on either method.
+ * https://www.paidmembershipspro.com/create-a-plugin-for-pmpro-customizations/
+ */
+
+ // Record when users gain the trial level.
+function one_time_trial_save_trial_level_used( $level_id, $user_id ) {
+	// Set this to the ID of your trial level.
+	$trial_level_id = 1; // Membership Level ID
+
+	if ( $level_id == $trial_level_id ) {
+		// Add user meta to record that the user has received their one-time trial.
+		update_user_meta( $user_id, 'pmpro_trial_level_used', $trial_level_id );
+	}
+}
+add_action( 'pmpro_after_change_membership_level', 'one_time_trial_save_trial_level_used', 10, 2 );
+
+// Show the user's trial meta setting for admins on the Edit Profile page.
+function one_time_trial_show_trial_level_used( $user ) {
+	if ( current_user_can( 'edit_users' ) ) { ?>
+		<h3>One-Time Trial</h3>
+		<table class="form-table"> 
+			<tbody>
+				<tr>
+					<th scope="row"></th>
+					<td>
+						<?php
+							$already = get_user_meta( $user->ID, 'pmpro_trial_level_used', true );
+						if ( ! empty( $already ) && $already == '1' ) {
+							echo 'Trial period has been claimed.';
+						} else {
+							echo 'Trial period not claimed.';
+						}
+						?>
+					</td>
+				</tr>
+			</tbody>
+		</table>
+		<?php
+	}
+}
+add_action( 'show_user_profile', 'one_time_trial_show_trial_level_used' );
+add_action( 'edit_user_profile', 'one_time_trial_show_trial_level_used' );
+
+// Check if the user has received their one-time trial at checkout.
+function one_time_trial_delay_pmpro_registration_checks() {
+	global $current_user;
+
+	//set this to the id of your trial level
+	$trial_level_id = 1; // Membership Level ID
+	if ( ! empty( $_REQUEST['level'] ) ) {
```

Comment:
This check needs to change, and any other instance of `$_REQUEST['level']` we can use the `pmpro_getLevelAtCheckout()` or from the $level (object) directly.